src/python/zquantum/vqe/singlet_uccsd.py

```diff
@@ -0,0 +1,131 @@
+from zquantum.core.interfaces.ansatz import Ansatz
+from zquantum.core.interfaces.ansatz_utils import (
+    ansatz_property,
+    invalidates_parametrized_circuit,
+)
+from zquantum.core.circuit import Circuit
+from openfermion.utils import uccsd_singlet_paramsize, uccsd_singlet_generator
+from overrides import overrides
+from .utils import exponentiate_fermion_operator, create_layer_of_gates
+from pyquil import Program
+from pyquil.gates import X
+from typing import List, Optional
+import sympy
+import numpy as np
+
+
+class SingletUCCSDAnsatz(Ansatz):
+
+    supports_parametrized_circuits = False
+    transformation = ansatz_property("transformation")
```

Comment:
So basically it is our own kind of `property`. It handles both assignment and retrieval of the value and additionally invalidates parametrized circuit of the ansatz (so that if you change value of this property, the parametrized circuit will get recomputed when you access it next time). You can think of it as a shortcut to the following code:
```python
    @property
    def transformation(self):
        return self._transformation

    @invalidates_parametrized_circuit
    @transformation.setter
    def transformation(self, transformation):
        self._transformation = transformation
```

The above 7 lines are therefore shortened to one. This pattern seems to be very common, so we decided to implement `ansatz_property` to reduce the amount of boilerplate.